ngraph/core/reference/include/ngraph/runtime/reference/ctc_greedy_decoder_seq_len.hpp

```diff
@@ -0,0 +1,73 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+#include <vector>
+#include "ngraph/coordinate_transform.hpp"
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename TF, typename TI, typename TCI, typename TSL>
+            void ctc_greedy_decoder_seq_len(const TF* data,
+                                            const TI* sequence_length,
+                                            const TI* blank_index,
+                                            TCI* out1,
+                                            TSL* out2,
+                                            const Shape& data_shape,
+                                            const Shape& out_shape,
+                                            const bool ctc_merge_repeated)
+            {
+                const auto batch_size = data_shape[0];
+                const auto class_count = data_shape[2];
+
+                const CoordinateTransform out_transform{out_shape};
```

Comment:
@ilyachur , @vladislav-volkov, @iefode, @jane-intel, @jdanieck, @ggalieroc 
Is `CoordinateTransform` allowed now in reference implementations?

Reply:
I propose to avoid it if it is possible.

Reply:
Please give me examples or suggestion for replacing CoordinateTransform

Reply:
A you can see in https://github.com/openvinotoolkit/openvino/pull/3633 we try to remove `CoordinateTransform` and calculate all in the fly. If you want to use something similar to `CooridianteTransform` look in to:
 https://github.com/openvinotoolkit/openvino/blob/bad4e97d9b9ea50146255acee9742572ff6baa47/ngraph/core/reference/src/runtime/reference/slice.cpp#L30

https://github.com/openvinotoolkit/openvino/blob/bad4e97d9b9ea50146255acee9742572ff6baa47/ngraph/core/reference/src/runtime/reference/reverse.cpp#L33